packages/api/src/common/proxies/city-validation-cache.proxy.ts

```diff
@@ -0,0 +1,31 @@
+import { CachePrefixes } from 'src/cache/enums/cache-prefixes.enum';
+import { CacheTTL } from 'src/cache/enums/cache-ttl.enum';
+import { CacheServiceInterface } from 'src/cache/interfaces/cache-service.interface';
+import { ChainableCityProvider } from 'src/city/interfaces/chainable-city.provider';
+
+export class CityProviderCacheProxy extends ChainableCityProvider {
+  private readonly keyBase: CachePrefixes = CachePrefixes.CITY_VALIDATION;
+  private readonly ttl: number = CacheTTL.MINUTES_10;
```

Comment:
Це дуже класна ідея з валідацією міста, але я помітив, що TTL для провалідованого міста 10 хвилин, але, мабуть, це якось знецінює ідею.

Бо ж якщо ми провалідуєм місто ```kyiv```, то продовж певного часу ми будем знати, що API може давати відповідь для ```kyiv``` і через це ми зменшимо кількість невдалих запитів, і тут час валідності міста має бути значно більший за дані про погоду для цього міста.

І якщо збільшити TTL, наприклад на годину, то продовж години ми будем знати, що всі запити з ```kyiv``` можна надсилати на API і вже потім кешувати response на 10 хвилин  

```suggestion
  private readonly ttl: number = CacheTTL.HOURS_1;
```

```suggestion
  export enum CacheTTL {
  MINUTES_10 = 10 * MINUTE,
  MINUTES_15 = 15 * MINUTE,
  HOURS_1 = 60 * MINUTE,
}```

Reply:
У мене валідація пов'язана з підпискою - отримання погоди йде окремо. Але додати перевірку хороша ідея.

Додав